contracts/0.8.9/StakingRouter.sol

```diff
@@ -299,27 +299,72 @@ contract StakingRouter is AccessControlEnumerable, BeaconChainDepositor, Version
+    /// `IStakingModule.getStakingModuleSummary()`. The overall process of updating validator counts is the following:
+    ///
+    /// 1. In the first data submission phase, the oracle calls `updateExitedValidatorsCountByStakingModule` on the
+    ///    staking router, passing the totals by module. The staking router stores these totals and uses them to
+    ///    distribute new stake and staking fees between the modules. There can only be single call of this function
+    ///    per oracle reporting frame.
+    ///
+    /// 2. In the first part of the second data submittion phase, the oracle calls
+    ///    `StakingRouter.reportStakingModuleStuckValidatorsCountByNodeOperator` on the staking router which passes the
+    ///    counts by node operator to the staking module by calling `IStakingModule.updateStuckValidatorsCount`.
+    ///    This can be done multiple times for the same module, passing data for different subsets of node operators.
+    ///
+    /// 3. In the second part of the second data submittion phase, the oracle calls
+    ///    `StakingRouter.reportStakingModuleExitedValidatorsCountByNodeOperator` on the staking router which passes
+    ///    the counts by node operator to the staking module by calling `IStakingModule.updateExitedValidatorsCount`.
+    ///    This can be done multiple times for the same module, passing data for different subsets of node
+    ///    operators.
+    ///
+    /// 4. At the end of the second data submission phase, it's expected for the aggragate exited validators count
+    ///    accross all module's node operators (stored in the module) to match the total count for this module
+    ///    (stored in the staking router). However, it might happen that the second phase of data submission doesn't
+    ///    finish until the new oracle reporting frame is started, in which case staking router will emit a warning
+    ///    event `StakingModuleExitedValidatorsIncompleteReporting` when the first data submission phase is performed
+    ///    for a new reporting frame. This condition will result in the staking module having an incomplete data about
+    ///    the exited and maybe stuck validator counts during the whole reporting frame. Handling this condition is
+    ///    the responsibility of each staking module.
+    ///
+    /// 5. When the second reporting phase is finshed, i.e. when the oracle submitted the complete data on the stuck
+    ///    and exited validator counts per node operator for the current reporting frame, the oracle calls
+    ///    `StakingRouter.onValidatorsCountsByNodeOperatorReportingFinished` which, in turn, calls
+    ///    `IStakingModule.onExitedAndStuckValidatorsCountsUpdated` on all modules.
+    ///
     function updateExitedValidatorsCountByStakingModule(
         uint256[] calldata _stakingModuleIds,
         uint256[] calldata _exitedValidatorsCounts
     )
         external
         onlyRole(REPORT_EXITED_VALIDATORS_ROLE)
+        returns (uint256)
```

Comment:
Super minor: `newlyExitedValidatorsCount` might be declared here, in the function declaration, then lines 355 and 386 might be removed